params/network_upgrades.go

```diff
@@ -5,18 +5,74 @@ package params
 
 import (
 	"math/big"
+
+	"github.com/ava-labs/avalanchego/utils/constants"
 )
 
-// NetworkUpgrades contains timestamps that enable avalanche network upgrades.
-type NetworkUpgrades struct {
-	SubnetEVMTimestamp *big.Int `json:"subnetEVMTimestamp,omitempty"` // A placeholder for the latest avalanche forks (nil = no fork, 0 = already activated)
+var (
+	LocalNetworkUpgrades = MandatoryNetworkUpgrades{
+		SubnetEVMTimestamp: big.NewInt(0),
+		DUpgradeTimestamp:  big.NewInt(0),
+	}
+
+	FujiNetworkUpgrades = MandatoryNetworkUpgrades{
+		SubnetEVMTimestamp: big.NewInt(0),
+		//	DUpgradeTimestamp:           big.NewInt(0), // TODO: Uncomment and set this to the correct value
+	}
+
+	MainnetNetworkUpgrades = MandatoryNetworkUpgrades{
+		SubnetEVMTimestamp: big.NewInt(0),
+		//	DUpgradeTimestamp:           big.NewInt(0), // TODO: Uncomment and set this to the correct value
+	}
+)
+
+// MandatoryNetworkUpgrades contains timestamps that enable mandatory network upgrades.
+// These upgrades are mandatory, meaning that if a node does not upgrade by the
+// specified timestamp, it will be unable to participate in consensus.
+// Avalanche specific network upgrades are also included here.
+type MandatoryNetworkUpgrades struct {
+	SubnetEVMTimestamp *big.Int `json:"subnetEVMTimestamp,omitempty"` // initial subnet-evm upgrade (nil = no fork, 0 = already activated)
+	DUpgradeTimestamp  *big.Int `json:"dUpgradeTimestamp,omitempty"`  // A placeholder for the latest avalanche forks (nil = no fork, 0 = already activated)
 }
 
-func (n *NetworkUpgrades) CheckCompatible(newcfg *NetworkUpgrades, headTimestamp *big.Int) *ConfigCompatError {
-	// Check subnet-evm specific activations
-	if isForkIncompatible(n.SubnetEVMTimestamp, newcfg.SubnetEVMTimestamp, headTimestamp) {
-		return newCompatError("SubnetEVM fork block timestamp", n.SubnetEVMTimestamp, newcfg.SubnetEVMTimestamp)
+func (m *MandatoryNetworkUpgrades) CheckMandatoryCompatible(newcfg *MandatoryNetworkUpgrades, headTimestamp *big.Int) *ConfigCompatError {
+	if isForkIncompatible(m.SubnetEVMTimestamp, newcfg.SubnetEVMTimestamp, headTimestamp) {
+		return newCompatError("SubnetEVM fork block timestamp", m.SubnetEVMTimestamp, newcfg.SubnetEVMTimestamp)
+	}
+	if isForkIncompatible(m.DUpgradeTimestamp, newcfg.DUpgradeTimestamp, headTimestamp) {
+		return newCompatError("DUpgrade fork block timestamp", m.DUpgradeTimestamp, newcfg.DUpgradeTimestamp)
 	}
+	return nil
+}
 
+func (m *MandatoryNetworkUpgrades) MandatoryForkOrder() []fork {
```

Comment:
Can we un-export this, so that we do not have an exported function returning an internal type?